code/modules/mob/living/simple_animal/bot/medbot.dm

```diff
@@ -361,65 +361,68 @@
 
 	return
 
+/mob/living/simple_animal/bot/medbot/proc/assess_beaker_injection(mob/living/carbon/C)
+	//If we have and are using a medicine beaker, return any reagent the patient is missing
+	if(use_beaker && reagent_glass?.reagents.total_volume)
+		for(var/datum/reagent/R in reagent_glass.reagents.reagent_list)
+			if(!C.reagents.has_reagent(R.id))
+				return R.id
+
+/mob/living/simple_animal/bot/medbot/proc/select_medication(mob/living/carbon/C, beaker_injection)
```

Comment:
Can you split this proc up into each attempt? Currently, each check is done in order anyway. Virus -> brute -> fire -> oxy -> tox is how the priority goes currently.